src/mcmc/hmc.jl

```diff
@@ -59,30 +61,23 @@ sample(gdemo([1.5, 2]), HMC(0.1, 10), 1000)
 sample(gdemo([1.5, 2]), HMC(0.01, 10), 1000)
 ```
 """
-struct HMC{AD, space, metricT <: AHMC.AbstractMetric} <: StaticHamiltonian{AD}
+struct HMC{AD, space, metricT <: AHMC.AbstractMetric} <: StaticHamiltonian
     ϵ::Float64 # leapfrog step size
     n_leapfrog::Int # leapfrog step number
+    adtype::AD
 end
 
-HMC(args...; kwargs...) = HMC{ADBackend()}(args...; kwargs...)
-function HMC{AD}(ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple) where {AD, metricT <: AHMC.AbstractMetric}
-    return HMC{AD, space, metricT}(ϵ, n_leapfrog)
-end
-function HMC{AD}(
-    ϵ::Float64,
-    n_leapfrog::Int,
-    ::Tuple{};
-    kwargs...
-) where AD
-    return HMC{AD}(ϵ, n_leapfrog; kwargs...)
+function HMC(ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple; adtype::ADTypes.AbstractADType = ADBackend()) where {metricT <: AHMC.AbstractMetric}
+    return HMC{typeof(adtype), space, metricT}(ϵ, n_leapfrog, adtype)
 end
-function HMC{AD}(
+function HMC(
     ϵ::Float64,
     n_leapfrog::Int,
     space::Symbol...;
-    metricT=AHMC.UnitEuclideanMetric
+    metricT=AHMC.UnitEuclideanMetric,
+    adtype::ADTypes.AbstractADType = ADBackend(),
 ) where AD
-    return HMC{AD}(ϵ, n_leapfrog, metricT, space)
+    return HMC(ϵ, n_leapfrog, metricT, space; adtype = adtype)
```

Comment:
And set the default AD type for each algorithm/sampler to a specific default such as AutoForwardDiff, you mean? Or would you like users to always specify the AD type explicitly?

Reply:
I would like to get rid of the global flag 👍 If we make it in a separate PR we should maybe wait with tagging a breaking release until this follow-up PR is merged as well, to avoid two breaking releases in a row.

Reply:
This is implemented in 0.30 ?? I updated turing and my get and error now. could you update the docs? in particular [autodiff](https://turinglang.org/v0.30/docs/using-turing/autodiff)